group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -2139,7 +2139,7 @@ public void onPartitionsDeleted(
         // At this point the metadata will not have been updated
         // with the deleted topics.
         Set<Uuid> topicIds = topicPartitions.stream()
-            .map(tp -> metadataImage.topics().getTopic(tp.topic()).id())
+            .map(tp -> metadataImage.topicId(tp.topic()).orElse(Uuid.ZERO_UUID))
```

Comment:
One thing that I was thinking about with the ZERO_UUID handling cases is that this group coordinator is only supported by clusters that use KRaft, and with KRaft you're guaranteed to have a topicId if one exists. So I'm not even sure there should be any cases where we want to return ZERO_UUID to clients?